nncf/common/accuracy_aware_training/training_loop.py

```diff
@@ -53,38 +54,28 @@ def run(self, model: TModel, train_epoch_fn, validate_fn, configure_optimizers_f
         (to be called inside the `train_epoch` of the TrainingRunner)
         :param validate_fn: a method to evaluate the model on the validation dataset
         (to be called inside the `train_epoch` of the TrainingRunner)
-        :param dump_checkpoint_fn: a method to dump a checkpoint
         :param configure_optimizers_fn: a method to instantiate an optimizer and a learning
         rate scheduler (to be called inside the `configure_optimizers` of the TrainingRunner)
+        :param dump_checkpoint_fn: a method to dump a checkpoint
+        :param load_checkpoint_fn: a method to load a checkpoint
+        :param early_stopping_fn: a method to check for an early stopping condition
         :return: The fine-tuned model
         """
 
 
-class EarlyExitCompressionTrainingLoop(TrainingLoop):
-    """
-    Adaptive compression training loop allows an accuracy-aware training process
-    to reach the maximal accuracy drop
-    (the maximal allowed accuracy degradation criterion is satisfied).
-    """
-
-    def __init__(self,
-                 nncf_config: NNCFConfig,
-                 compression_controller: CompressionAlgorithmController,
-                 lr_updates_needed=True, verbose=True,
-                 dump_checkpoints=True):
-        super().__init__()
-        accuracy_aware_training_params = extract_accuracy_aware_training_params(nncf_config)
-        runner_factory = EarlyExitTrainingRunnerCreator(accuracy_aware_training_params,
-                                                        compression_controller,
-                                                        lr_updates_needed, verbose,
-                                                        dump_checkpoints)
-        self.runner = runner_factory.create_training_loop()
-        self.compression_controller = compression_controller
+class BaseEarlyExitCompressionTrainingLoop(TrainingLoop, ABC):
+    def __init__(self):
+        self.runner = None
+        self.compression_controller = None
 
-    def run(self, model, train_epoch_fn, validate_fn, configure_optimizers_fn=None,
-            dump_checkpoint_fn=None, tensorboard_writer=None, log_dir=None):
+    def run(self, model: TModel, train_epoch_fn, validate_fn, configure_optimizers_fn=None,
```

Comment:
This could use more type hints

Reply:
It was decided to improve code quality within a separate task